//specify the proto version
syntax = "proto3";
//the name(or empty)
package groupChat;

service ChatServer {
    rpc chatFunction(stream ChatInput) returns (stream ChatOutput) {}
}

message ChatInput {
    //type 1: login, 2: join, 3: chat, 4: like, 5: dislike, 6: history
    int32 type = 1;
    string message = 2;
}

message ChatOutput {
    string status = 1;
    repeated ChatMessage messages = 2;
}

message ChatMessage{
    int32 id = 1;
    string content = 2;
    int32 numberOfLikes = 3;
}